packages/editor/src/envelope/LoadingScreen/LoadingScreen.tsx

```diff
@@ -15,48 +15,54 @@
  */
 
 import * as React from "react";
-import { useLayoutEffect, useState } from "react";
 import { Bullseye, Page, Spinner, Title } from "@patternfly/react-core";
 import "./styles.scss";
-import { useEditorEnvelopeI18nContext } from "../i18n/setup";
+import { useEditorEnvelopeI18nContext } from "../i18n";
+import { useCallback, useLayoutEffect, useMemo, useState } from "react";
 
-export const FADE_OUT_DELAY = 400;
-
-export function LoadingScreen(props: { visible: boolean }) {
-  let cssAnimation;
+export function LoadingScreen(props: { loading: boolean }) {
   const [mustRender, setMustRender] = useState(true);
   const { i18n } = useEditorEnvelopeI18nContext();
 
-  if (props.visible) {
-    cssAnimation = { opacity: 1 };
-  } else {
-    cssAnimation = { opacity: 0, transition: `opacity ${FADE_OUT_DELAY}ms` };
-  }
+  const onAnimationEnd = useCallback((e: React.AnimationEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    setMustRender(false);
+  }, []);
+
+  const loadingScreenClassName = useMemo(() => {
+    if (props.loading) {
+      return "";
+    } else {
+      return "loading-finished";
+    }
```

Comment:
```suggestion
   return props.loading ? "" : "loading-finished";
```
A suggestion to make it shorter.

Reply:
It's shorter, but I think it can be trickier to read. I left like this to be easier, but I should've removed the else clause.  :grimacing: